pallets/aliases/src/lib.rs

```diff
@@ -0,0 +1,379 @@
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// The price (in native NOVA) for a Namespace
+		#[pallet::constant]
+		type NamespacePrice: Get<u128>;
+
+		/// The max size of namespace string
+		#[pallet::constant]
+		type NameLimit: Get<u32>;
+	}
+
+	/// **StorageMap** that maps a **Namespace** to its owner.
+	#[pallet::storage]
+	pub type Namespaces<T: Config> = StorageMap<_, Twox64Concat, Vec<u8>, T::AccountId>;
+
+	/// **StorageMap** that maps a **Name (of type `Vec<u8>`)** to an **index**.
+	/// This ensures no duplicated aliases are used.
+	#[pallet::storage]
+	pub type Names<T: Config> = StorageMap<_, Twox64Concat, Vec<u8>, Compact<u64>>;
+
+	/// **StorageValue** that **equals** the **total number of unique names** used for aliases in the chain.
+	#[pallet::storage]
+	pub type NamesIndex<T: Config> = StorageValue<_, u64, ValueQuery>;
+
+	/// **StorageDoubleMap** that maps a (**Namespace**, **Alias** name index) with a **LinkTarget**.
+	#[pallet::storage]
+	pub type Aliases<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		Vec<u8>, // namespace
+		Twox64Concat,
+		Compact<u64>, // alias name index
+		LinkTarget<T::AccountId>, // target asset
+	>;
+
+	#[allow(missing_docs)]
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		NamespaceCreated {
+			who: T::AccountId,
+			namespace: Vec<u8>,
+		},
+		NamespaceTransferred {
+			namespace: Vec<u8>,
+			from: T::AccountId,
+			to: T::AccountId,
+		},
+		NamespaceDeleted {
+			namespace: Vec<u8>,
+		},
+	}
+
+	// Errors inform users that something went wrong.
+	#[allow(missing_docs)]
+	#[pallet::error]
+	pub enum Error<T> {
+		InvalidInput,
+		NamespaceAlreadyExists,
+		NamespaceNotFound,
+		NotAllowed,
+		SystematicFailure,
+		AliasAlreadyOwned,
+		AliasNotExists,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Create a new **Namespace**, a unique string that identifies an owner of assets.
+		///
+		/// The creation of a namespace burns NOVA from the caller's balance.
+		/// The amount is set in Config `NamespacePrice`.
+		///
+		/// - `namespace`: namespace to create
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+		pub fn create_namespace(
+			origin: OriginFor<T>,
+			namespace: BoundedVec<u8, <T as pallet::Config>::NameLimit>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+
+			ensure!(!namespace.len().is_zero(), Error::<T>::InvalidInput);
+
+			let namespace = namespace.into_inner();
+
+			// Check that the namespace does not exist already
+			ensure!(!<Namespaces<T>>::contains_key(&namespace), Error::<T>::NamespaceAlreadyExists);
+
+			// burn NOVA from account's balance. The amount is set in Config.
+			let amount: <T as pallet_balances::Config>::Balance =
+				<T as Config>::NamespacePrice::get().saturated_into();
+			<pallet_balances::Pallet<T> as fungible::Mutate<T::AccountId>>::burn_from(
```

Comment:
I'm not sure if burn is correct, I think it's `withdraw`, this should be just like a transaction basically, feel free to dig about it.

Reply:
Right, it's withdraw to reduce account's balance. Burn would reduce the issuance. Thanks!